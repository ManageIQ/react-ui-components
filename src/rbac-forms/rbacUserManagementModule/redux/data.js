export const data = {
  rows: [
    {
      id: 100,
      name: 'Adam the first',
      userid: 'Admin',
      email: 'email@neco.com',
      current_group: {
        label: 'Some group',
        onClick: () => console.log('Current group clicked'),
      },
      role: {
        label: 'Admin role',
        onClick: () => console.log('Current role click'),
      },
      groups: [{
        label: 'Cloud-Operators',
        icon: 'group',
        groupId: 1,
        onClick: () => console.log('Group click'),
      }, {
        label: 'Cloud-Users',
        icon: 'group',
        groupId: 2,
        onClick: () => console.log('Group click'),
      }],
      lastlogon: '06/25/18 06:58:14 UTC',
      lastlogoff: '06/30/18 06:58:14 UTC',
    },
    {
      id: 101,
      name: 'Marge the middle',
      userid: 'La transparenza della qualita',
      email: 'email@jandak.com',
      current_group: {
        label: 'Mediocre group',
        onClick: () => console.log('Current group clicked'),
      },
      role: {
        label: 'Shot role',
        onClick: () => console.log('Current role click'),
      },
      groups: [{
        label: 'Cloud-Operators',
        icon: 'group',
        groupId: 1,
        onClick: () => console.log('Group click'),
      }, {
        label: 'Cloud-Users',
        icon: 'group',
        groupId: 2,
        onClick: () => console.log('Group click'),
      }],
      lastlogon: '01/25/18 06:58:14 UTC',
      lastlogoff: '06/30/18 06:58:14 UTC',
    },
    {
      id: 102,
      name: 'Zlatko the last',
      userid: 'Peligro',
      email: 'email@jandak.com',
      current_group: {
        label: 'Admin group',
        onClick: () => console.log('Current group clicked'),
      },
      role: {
        label: 'Cloud-Users',
        onClick: () => console.log('Current role click'),
      },
      groups: [{
        label: 'Cloud-Operators',
        icon: 'group',
        groupId: 1,
        onClick: () => console.log('Group click'),
      }, {
        label: 'Cloud-Users',
        icon: 'group',
        groupId: 2,
        onClick: () => console.log('Group click'),
      }],
      lastlogon: '08/25/18 06:58:14 UTC',
      lastlogoff: '010/30/18 06:58:14 UTC',
    },
  ],
  columns: [{
    property: 'name',
    label: 'Full Name',
  }, {
    property: 'userid',
    label: 'Username',
  }, {
    property: 'email',
    label: 'E-mail',
  }, {
    property: 'current_group',
    label: 'Current Group',
  }, {
    property: 'role',
    label: 'Role',
  }, {
    property: 'lastlogon',
    label: 'Last Logon',
  }, {
    property: 'lastlogoff',
    label: 'Last Logoff',
  }],
  selectedUsers: null,
  groups: [
    {
      value: 1,
      label: 'Hope',
    },
    {
      value: 2,
      label: 'Christa',
    },
    {
      value: 3,
      label: 'Tania',
    },
    {
      value: 4,
      label: 'Mcintyre',
    },
    {
      value: 5,
      label: 'Tonya',
    },
    {
      value: 6,
      label: 'Marcie',
    },
    {
      value: 7,
      label: 'Cara',
    },
    {
      value: 8,
      label: 'Becker',
    },
    {
      value: 9,
      label: 'Foley',
    },
    {
      value: 10,
      label: 'Perkins',
    },
  ],
};
