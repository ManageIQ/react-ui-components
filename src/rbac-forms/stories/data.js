import { action } from '@storybook/addon-actions';

export const groups = [
  {
    value: 1,
    label: 'Hope',
  },
  {
    value: 2,
    label: 'Christa',
  },
  {
    value: 3,
    label: 'Tania',
  },
  {
    value: 4,
    label: 'Mcintyre',
  },
  {
    value: 5,
    label: 'Tonya',
  },
  {
    value: 6,
    label: 'Marcie',
  },
  {
    value: 7,
    label: 'Cara',
  },
  {
    value: 8,
    label: 'Becker',
  },
  {
    value: 9,
    label: 'Foley',
  },
  {
    value: 10,
    label: 'Perkins',
  },
];

export const editedUser = {
  name: 'Users full name',
  userid: 'admin jiny',
  chosen_group: [5, 7],
};

export const usersTableColumns = [{
  property: 'fullname',
  label: 'Full Name',
}, {
  property: 'username',
  label: 'Username',
}, {
  property: 'email',
  label: 'E-mail',
}, {
  property: 'currentgroup',
  label: 'Current Group',
}, {
  property: 'role',
  label: 'Role',
}, {
  property: 'lastlogon',
  label: 'Last Logon',
}, {
  property: 'lastlogoff',
  label: 'Last Logoff',
}];

export const usersTableRows = [
  {
    id: 100,
    fullname: 'Adam the first',
    username: 'Admin',
    email: 'email@neco.com',
    currentgroup: 'Some group',
    role: 'Admin role',
    lastlogon: '06/25/18 06:58:14 UTC',
    lastlogoff: '06/30/18 06:58:14 UTC',
  },
  {
    id: 101,
    fullname: 'Marge the middle',
    username: 'La transparenza della qualita',
    email: 'email@jandak.com',
    currentgroup: 'Mediocre group',
    role: 'Shot role',
    lastlogon: '01/25/18 06:58:14 UTC',
    lastlogoff: '06/30/18 06:58:14 UTC',
  },
  {
    id: 102,
    fullname: 'Zlatko the last',
    username: 'Peligro',
    email: 'email@jandak.com',
    currentgroup: 'Super group',
    role: 'Shot role',
    lastlogon: '08/25/18 06:58:14 UTC',
    lastlogoff: '010/30/18 06:58:14 UTC',
  },
];

export const user = {
  name: 'Administrator',
  userid: 'Admin',
  email: 'mail@mail.com',
  current_group: {
    label: 'EvmGroup-super_administrator',
    onClick: () => action('Current group clicked'),
  },
  groups: [{
    label: 'Cloud-Operators',
    icon: 'group',
    groupId: 1,
    onClick: () => action('Group click'),
  }, {
    label: 'Cloud-Users',
    icon: 'group',
    groupId: 2,
    onClick: () => action('Group click'),
  }],
  role: {
    label: 'Cloud-Users',
    onClick: () => action('Current role click'),
  },
};
