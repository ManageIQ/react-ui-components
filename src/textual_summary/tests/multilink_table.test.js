import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import MultilinkTable from '../multilink_table';
import { multilinkTableData } from '../data/multilink_table';

describe('MultilinkTable', () => {
  it('renders just fine', () => {
    const table = shallow(<MultilinkTable
      title={multilinkTableData.title}
      items={multilinkTableData.items}
      onClick={() => null}
    />);
    expect(toJson(table)).toMatchSnapshot();
  });
});
