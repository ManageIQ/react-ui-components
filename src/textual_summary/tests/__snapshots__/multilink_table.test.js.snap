// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MultilinkTable renders just fine 1`] = `
<table
  className="table table-bordered table-striped table-summary-screen"
>
  <thead>
    <tr>
      <th
        colSpan="100"
      >
        Some OpenStack Stuff
      </th>
    </tr>
  </thead>
  <tbody>
    <tr
      className="no-hover"
      key="0"
    >
      <td
        className="hover"
      >
        <IconOrImage
          icon={null}
          image={null}
          title={null}
        />
        Aodh
      </td>
      <td
        className="hover"
        key="0"
        onClick={[Function]}
        style={
          Object {
            "cursor": "pointer, hand",
          }
        }
        title="Show list of running Aodh"
      >
        <IconOrImage
          icon="pficon pficon-ok"
          image={null}
          title="Show list of running Aodh"
        />
        Running (3)
      </td>
      <td
        key="1"
        title="Show list of failed Aodh"
      >
        <IconOrImage
          icon={null}
          image={null}
          title="Show list of failed Aodh"
        />
        Failed (0)
      </td>
      <td
        className="hover"
        key="2"
        onClick={[Function]}
        style={
          Object {
            "cursor": "pointer, hand",
          }
        }
        title="Show list of all Aodh"
      >
        <IconOrImage
          icon="pficon pficon-service"
          image={null}
          title="Show list of all Aodh"
        />
        All (3)
      </td>
      <td
        key="3"
        title="Show list of configuration files of Aodh"
      >
        <IconOrImage
          icon="fa fa-file-o"
          image={null}
          title="Show list of configuration files of Aodh"
        />
        Configuration (0)
      </td>
    </tr>
    <tr
      className="no-hover"
      key="1"
    >
      <td
        className="hover"
      >
        <IconOrImage
          icon={null}
          image={null}
          title={null}
        />
        Ceilometer
      </td>
      <td
        className="hover"
        key="0"
        onClick={[Function]}
        style={
          Object {
            "cursor": "pointer, hand",
          }
        }
        title="Show list of running Ceilometer"
      >
        <IconOrImage
          icon="pficon pficon-ok"
          image={null}
          title="Show list of running Ceilometer"
        />
        Running (3)
      </td>
      <td
        key="1"
        title="Show list of failed Ceilometer"
      >
        <IconOrImage
          icon={null}
          image={null}
          title="Show list of failed Ceilometer"
        />
        Failed (0)
      </td>
      <td
        className="hover"
        key="2"
        onClick={[Function]}
        style={
          Object {
            "cursor": "pointer, hand",
          }
        }
        title="Show list of all Ceilometer"
      >
        <IconOrImage
          icon="pficon pficon-service"
          image={null}
          title="Show list of all Ceilometer"
        />
        All (3)
      </td>
      <td
        key="3"
        title="Show list of configuration files of Ceilometer"
      >
        <IconOrImage
          icon="fa fa-file-o"
          image={null}
          title="Show list of configuration files of Ceilometer"
        />
        Configuration (0)
      </td>
    </tr>
    <tr
      className="no-hover"
      key="2"
    >
      <td
        className="hover"
      >
        <IconOrImage
          icon={null}
          image={null}
          title={null}
        />
        Cinder
      </td>
      <td
        className="hover"
        key="0"
        onClick={[Function]}
        style={
          Object {
            "cursor": "pointer, hand",
          }
        }
        title="Show list of running Cinder"
      >
        <IconOrImage
          icon="pficon pficon-ok"
          image={null}
          title="Show list of running Cinder"
        />
        Running (3)
      </td>
      <td
        key="1"
        title="Show list of failed Cinder"
      >
        <IconOrImage
          icon={null}
          image={null}
          title="Show list of failed Cinder"
        />
        Failed (0)
      </td>
      <td
        className="hover"
        key="2"
        onClick={[Function]}
        style={
          Object {
            "cursor": "pointer, hand",
          }
        }
        title="Show list of all Cinder"
      >
        <IconOrImage
          icon="pficon pficon-service"
          image={null}
          title="Show list of all Cinder"
        />
        All (3)
      </td>
      <td
        key="3"
        title="Show list of configuration files of Cinder"
      >
        <IconOrImage
          icon="fa fa-file-o"
          image={null}
          title="Show list of configuration files of Cinder"
        />
        Configuration (0)
      </td>
    </tr>
    <tr
      className="no-hover"
      key="3"
    >
      <td
        className="hover"
      >
        <IconOrImage
          icon={null}
          image={null}
          title={null}
        />
        Glance
      </td>
      <td
        className="hover"
        key="0"
        onClick={[Function]}
        style={
          Object {
            "cursor": "pointer, hand",
          }
        }
        title="Show list of running Glance"
      >
        <IconOrImage
          icon="pficon pficon-ok"
          image={null}
          title="Show list of running Glance"
        />
        Running (2)
      </td>
      <td
        key="1"
        title="Show list of failed Glance"
      >
        <IconOrImage
          icon={null}
          image={null}
          title="Show list of failed Glance"
        />
        Failed (0)
      </td>
      <td
        className="hover"
        key="2"
        onClick={[Function]}
        style={
          Object {
            "cursor": "pointer, hand",
          }
        }
        title="Show list of all Glance"
      >
        <IconOrImage
          icon="pficon pficon-service"
          image={null}
          title="Show list of all Glance"
        />
        All (2)
      </td>
      <td
        key="3"
        title="Show list of configuration files of Glance"
      >
        <IconOrImage
          icon="fa fa-file-o"
          image={null}
          title="Show list of configuration files of Glance"
        />
        Configuration (0)
      </td>
    </tr>
    <tr
      className="no-hover"
      key="4"
    >
      <td
        className="hover"
      >
        <IconOrImage
          icon={null}
          image={null}
          title={null}
        />
        Gnocchi
      </td>
      <td
        className="hover"
        key="0"
        onClick={[Function]}
        style={
          Object {
            "cursor": "pointer, hand",
          }
        }
        title="Show list of running Gnocchi"
      >
        <IconOrImage
          icon="pficon pficon-ok"
          image={null}
          title="Show list of running Gnocchi"
        />
        Running (2)
      </td>
      <td
        key="1"
        title="Show list of failed Gnocchi"
      >
        <IconOrImage
          icon={null}
          image={null}
          title="Show list of failed Gnocchi"
        />
        Failed (0)
      </td>
      <td
        className="hover"
        key="2"
        onClick={[Function]}
        style={
          Object {
            "cursor": "pointer, hand",
          }
        }
        title="Show list of all Gnocchi"
      >
        <IconOrImage
          icon="pficon pficon-service"
          image={null}
          title="Show list of all Gnocchi"
        />
        All (2)
      </td>
      <td
        key="3"
        title="Show list of configuration files of Gnocchi"
      >
        <IconOrImage
          icon="fa fa-file-o"
          image={null}
          title="Show list of configuration files of Gnocchi"
        />
        Configuration (0)
      </td>
    </tr>
    <tr
      className="no-hover"
      key="5"
    >
      <td
        className="hover"
      >
        <IconOrImage
          icon={null}
          image={null}
          title={null}
        />
        Heat
      </td>
      <td
        className="hover"
        key="0"
        onClick={[Function]}
        style={
          Object {
            "cursor": "pointer, hand",
          }
        }
        title="Show list of running Heat"
      >
        <IconOrImage
          icon="pficon pficon-ok"
          image={null}
          title="Show list of running Heat"
        />
        Running (4)
      </td>
      <td
        key="1"
        title="Show list of failed Heat"
      >
        <IconOrImage
          icon={null}
          image={null}
          title="Show list of failed Heat"
        />
        Failed (0)
      </td>
      <td
        className="hover"
        key="2"
        onClick={[Function]}
        style={
          Object {
            "cursor": "pointer, hand",
          }
        }
        title="Show list of all Heat"
      >
        <IconOrImage
          icon="pficon pficon-service"
          image={null}
          title="Show list of all Heat"
        />
        All (4)
      </td>
      <td
        key="3"
        title="Show list of configuration files of Heat"
      >
        <IconOrImage
          icon="fa fa-file-o"
          image={null}
          title="Show list of configuration files of Heat"
        />
        Configuration (0)
      </td>
    </tr>
    <tr
      className="no-hover"
      key="6"
    >
      <td
        className="hover"
      >
        <IconOrImage
          icon={null}
          image={null}
          title={null}
        />
        Keystone
      </td>
      <td
        key="0"
        title="Show list of running Keystone"
      >
        <IconOrImage
          icon={null}
          image={null}
          title="Show list of running Keystone"
        />
        Running (0)
      </td>
      <td
        key="1"
        title="Show list of failed Keystone"
      >
        <IconOrImage
          icon={null}
          image={null}
          title="Show list of failed Keystone"
        />
        Failed (0)
      </td>
      <td
        className="hover"
        key="2"
        onClick={[Function]}
        style={
          Object {
            "cursor": "pointer, hand",
          }
        }
        title="Show list of all Keystone"
      >
        <IconOrImage
          icon="pficon pficon-service"
          image={null}
          title="Show list of all Keystone"
        />
        All (1)
      </td>
      <td
        key="3"
        title="Show list of configuration files of Keystone"
      >
        <IconOrImage
          icon="fa fa-file-o"
          image={null}
          title="Show list of configuration files of Keystone"
        />
        Configuration (0)
      </td>
    </tr>
    <tr
      className="no-hover"
      key="7"
    >
      <td
        className="hover"
      >
        <IconOrImage
          icon={null}
          image={null}
          title={null}
        />
        Nova
      </td>
      <td
        className="hover"
        key="0"
        onClick={[Function]}
        style={
          Object {
            "cursor": "pointer, hand",
          }
        }
        title="Show list of running Nova"
      >
        <IconOrImage
          icon="pficon pficon-ok"
          image={null}
          title="Show list of running Nova"
        />
        Running (5)
      </td>
      <td
        key="1"
        title="Show list of failed Nova"
      >
        <IconOrImage
          icon={null}
          image={null}
          title="Show list of failed Nova"
        />
        Failed (0)
      </td>
      <td
        className="hover"
        key="2"
        onClick={[Function]}
        style={
          Object {
            "cursor": "pointer, hand",
          }
        }
        title="Show list of all Nova"
      >
        <IconOrImage
          icon="pficon pficon-service"
          image={null}
          title="Show list of all Nova"
        />
        All (10)
      </td>
      <td
        key="3"
        title="Show list of configuration files of Nova"
      >
        <IconOrImage
          icon="fa fa-file-o"
          image={null}
          title="Show list of configuration files of Nova"
        />
        Configuration (0)
      </td>
    </tr>
    <tr
      className="no-hover"
      key="8"
    >
      <td
        className="hover"
      >
        <IconOrImage
          icon={null}
          image={null}
          title={null}
        />
        Swift
      </td>
      <td
        className="hover"
        key="0"
        onClick={[Function]}
        style={
          Object {
            "cursor": "pointer, hand",
          }
        }
        title="Show list of running Swift"
      >
        <IconOrImage
          icon="pficon pficon-ok"
          image={null}
          title="Show list of running Swift"
        />
        Running (14)
      </td>
      <td
        key="1"
        title="Show list of failed Swift"
      >
        <IconOrImage
          icon={null}
          image={null}
          title="Show list of failed Swift"
        />
        Failed (0)
      </td>
      <td
        className="hover"
        key="2"
        onClick={[Function]}
        style={
          Object {
            "cursor": "pointer, hand",
          }
        }
        title="Show list of all Swift"
      >
        <IconOrImage
          icon="pficon pficon-service"
          image={null}
          title="Show list of all Swift"
        />
        All (14)
      </td>
      <td
        key="3"
        title="Show list of configuration files of Swift"
      >
        <IconOrImage
          icon="fa fa-file-o"
          image={null}
          title="Show list of configuration files of Swift"
        />
        Configuration (0)
      </td>
    </tr>
  </tbody>
</table>
`;
